cmp/form/JsonField.js

```diff
@@ -27,17 +27,27 @@ import './JsonField.css';
  *   A JSON Editor
  *
  * @prop rest, see general properties for HoistField
+ *      Accepts props documented below as well as any properties supported by the CodeMirror api,
+ *      with the exception of 'value', which is managed by HoistField.
+ *      Provided CodeMirror props will override any editor properties preconfigured here.
+ *      See: https://codemirror.net/doc/manual.html#api_configuration for details.
  *
  * @prop width, width of field, in pixels
  * @prop height, width of field, in pixels
- * @prop lineWrapping, Whether field should scroll or wrap for long lines. Defaults to false
+ * @prop disabled, Whether the field is readonly, Defaults to false
  */
 @hoistComponent()
 export class JsonField extends HoistField {
 
     editor = null;
     taCmp = null;
 
+    static defaultProps = {
+        width: null,
```

Comment:
Looking at the code mirror source it appears that an undefined height or width will be handled the same as null in the setSize method, but in the docs it suggests null. Including here allows us to follow the docs, be more self documenting, and avoids having to use 'let' in the implementation to allow conversation as we were doing before.

https://github.com/codemirror/CodeMirror/blob/2c741bd6742678f67e812307b2f78db4f4566d9d/src/edit/methods.js#L392
cm.setSize(width: number|string, height: number|string)
Programmatically set the size of the editor (overriding the applicable CSS rules). width and height can be either numbers (interpreted as pixels) or CSS units ("100%", for example). You can pass null for either of them to indicate that that dimension should not be changed.